### PR TITLE
tweaks to look and feel of autodoc

### PR DESCRIPTION
**Pull request by nicolaskruchten**
Here's a quick pass of suggested changes to the autodoc layout.

### CODE REVIEW

**Comment by emmanuelle** [comment]
My only concern is that the methods of `go.Figure` have disappeared.

**Comment by nicolaskruchten** [comment]
Oh that's weird... when I was rendering this last night it wasn't doing this.

**Comment by nicolaskruchten** [comment]
I think there was some `make` caching happening such that things looked great after `make html` but not after `make clean` :(

File: apidoc/graph_objects.rst [hunk 1]
```diff
@@ -11,12 +11,14 @@ Graph objects: figures, traces and layout
...
-.. toctree::
-   :maxdepth: 1
-   :caption: Figure:
+Figure
+------
 
-   figure.rst
+.. autosummary::
+   :toctree: generated/
+   :template: trace.rst
```
Comment (nicolaskruchten) [inline apidoc/graph_objects.rst]: @emmanuelle is this the line that's preventing the `Figure` methods from appearing? should we have a `Figure`-specific template that makes them appear so we can keep the rest of this formatting in place?

Comment (emmanuelle) [reply apidoc/graph_objects.rst]: Yes, I'm working on it, can I push a commit directly to your branch? We need the template to include the members and inherited-members (which was done in the `figure.rst` file).

Comment (nicolaskruchten) [reply apidoc/graph_objects.rst]: Awesome, and yes, feel free to push to this branch!

**Comment by emmanuelle** [comment]
I just spent 2 hours trying all possible options with autoclass and autosummary... I pushed to #188 an alternative solution which exposes methods of Figure, and has everything in `generated` (but we lose the toc with `update_layout` etc., it seems that I can't get all of this to work together :-()

**Pull request closed without merging**